File: 987-reveal-cards-in-increasing-order/reveal-cards-in-increasing-order.cpp
class Solution {
public:
    vector<int> deckRevealedIncreasing(vector<int>& deck) {
        queue<int> q;
        int n=deck.size();
        sort(deck.begin(),deck.end());
        for(int i=0;i<n;i++){
            q.push(i);
        }
        int i=0;
        vector<int>nums(n,0);
        while(!q.empty()){
            int idx=q.front();
            q.pop();
            int v=q.front();
            q.pop();
            q.push(v);
            nums[idx]=deck[i];
            i++;
        }
        return nums;
    }
};

File: 987-reveal-cards-in-increasing-order/reveal-cards-in-increasing-order.py
class Solution:
    def deckRevealedIncreasing(self, deck: List[int]) -> List[int]:
        n=len(deck)
        deck.sort()
        q=deque()
        for i in range(n):
            q.append(i)
        i=0
        nums=[0]*n
        if len(q)==0:
            return []
        while len(q)>0:
            idx=q.popleft()
            nums[idx]=deck[i]
            i+=1
            if len(q)>0:
                q.append(q.popleft())
        return nums
